implementation/rust-embedded-library.tex

```diff
@@ -1,65 +1,62 @@
-\section{Rust Embedded Library}
-\label{sec:rust_embedded_library}
+\section{The Rust Embedded Library}
+\label{sec:rel}
 
-When we consider {\std}'s dependencies, shown in \autoref{fig:rust:librust}, we see that the crate is not suitable for a bare-metal application because it depends on an \gls{os} to be present.
-The dependency on either a Unix or a Windows \gls{os} excludes {\std} and all of its dependents from being used directly with the EFM32.
+As mentioned in \autoref{sec:rsl}, \gls{rsl} is built with the assumtion that the application is running on an OS.
 This section presents some of the libraries provided by {\rust} that can be used in a bare-metal system, and how and why they are composed into what we refer to as the \gls{rel}.
+\gls{rel} is unlike \gls{rsl} not built as a facade, in fact it is not implemented at all.
+It it just a way to provide a well defined definition of the {\rust} library in an embedded system.
 
 \begin{figure}[H]
   \begin{center}
     \includegraphics[scale=0.3]{figures/background/rust/embedded-rust-lib.png}
   \end{center}
-  \caption{\rust Embedded Library}
+  \caption{{\rust} Embedded Library}
   \label{fig:rust:rel}
 \end{figure}
 
-\subsection{Allocation}
-\label{sec:rust:allocation}
-
-The \gls{rcl} does not use or expose any heap allocation.
-Heap allocation is introduced through the \code{Box} type defined in the allocation library.
+\subsection{The Core Library}
+\label{sec:rust:core}
 
-\subsection{The Collection Library}
-
-The {\rust} collection library provides the data structures shown in \autoref{tab:rust:collections}.
+As described in \autoref{sec:rcl} the \gls{rcl} defines the \emph{core} functionality of the {\rust} language.
+These includes the primitive types listed in \autoref{tab:rust:datatypes} in addition to some key datastructures as \code{Option} and \code{Result} as described in \autoref{ssub:rust:features}.
+\gls{rcl} defines functionality to work with atomic values and cells along with exposing compiler \concept{intrinsics} which are used for example to provide \concept{volatile} access to memory.
+The \gls{rcl} does not have any library dependencies, but in order to use the library without \gls{rsl} a few definitions given in \autoref{tab:core:definitions} must exists.
 
 \begin{table}[H]
-  \begin{tabular}{l}
-    Binary Heap \\
-    Binary Tree Map \\
-    Binary Tree Set \\
-    Bit Set \\
-    Bit Vector \\
-    Enum Set \\
-    Linked List \\
-    Vector \\
-    Vector Dequeue \\
-    Vector Map \\
-    String \\
+  \begin{tabular}{l l}
+    \code{memcpy, memcmp, memset} & basic memory management \\
+    \code{rust\_begin\_unwind} & handling panicing \\
```

Comment:
Kanskje det passer med en setning her om hva en panic er?

f.eks. `Handles panicking. A panic is Rust's way of unwinding the currently executing thread, evidently resulting in the thread being terminated` eller noe ligenende.